app/src/main/java/net/squanchy/signin/SignInService.kt

```diff
@@ -28,15 +26,15 @@ class SignInService(private val authService: FirebaseAuthService) {
                     currentUser()
                         .firstOrError()
                         .flatMapCompletable { Completable.complete() }
+                } else {
+                    authService.signInAnonymously()
                 }
-
-                authService.signInAnonymously()
             }
     }
 
     fun currentUser(): Observable<Optional<FirebaseUser>> {
         return authService.currentUser()
-            .subscribeOn(Schedulers.io())
+            //.subscribeOn(Schedulers.io())
```

Comment:
🤔

Reply:
Forgot to uncomment, we were trying to see if it was some other weird threading issue with Firebase :D

Reply:
yeah damn